pkg/infoschema/test/infoschemav2test/v2_test.go

```diff
@@ -533,3 +533,39 @@ func TestSnapshotInfoschemaReader(t *testing.T) {
 	sql = fmt.Sprintf("select * from INFORMATION_SCHEMA.TABLES as of timestamp '%s' where table_schema = 'issue55827'", timeStr)
 	tk.MustQuery(sql).Check(testkit.Rows())
 }
+
+
+func TestIssue55835(t *testing.T) {
```

Comment:
The UT can pass without this fix

Reply:
The problem is, when we switch v1/v2 back and forth, the *history* is not cleared.
So v1/v2 switch can't reproduce the tidb restart scenario.
During adding this test, I did find something wrong: when modify @@tidb_schema_cache_size, without a new DDL operation, v1/v2 switch does not happen, that's not expected.

I confirmed this change can fix the bug and tested manually following the steps described in the issue.